### ISSUE TITLE
Bug: POST method caching not working after version 0.10.7

### ISSUE THREAD

**Issue by subhradeepnag** [arthurfiorette/axios-cache-interceptor#546]
Hi again!

So I noticed that POST-caching has stopped working after version 0.10.7.

For example this code

```
`const Axios = require("axios");
const { setupCache } = require("axios-cache-interceptor/dev");
const cacheTTL = 30 * 1000 // 30 seconds
const axiosInstance = Axios.create()
const cacheOptions = {
  ttl: cacheTTL,
  interpretHeader: false,
  methods: ['post',],
  debug: console.log
}
const axiosCached = setupCache(axiosInstance, cacheOptions)

axiosCached.post("https://reqres.in/api/users",{
      name: "morpheus",
      job: "leader",
    } ).then((res) => {
    console.log(res.cached);
  }).catch((err) => {
    console.log(err);
  });
  axiosCached.post("https://reqres.in/api/users",{
        name: "morpheus",
        job: "leader",
      })
    .then((res) => {
      console.log(res.cached);
    })
    .catch((err) => {
      console.log(err);
    });

`
```

If we try this with axios-cache-interceptor version 0.10.7, it caches it properly. But if we try it any version after that it does not cache.

**Comment by subhradeepnag** [comment]
Screenshots - Version 0.10.7

<img width="806" alt="Screenshot 2023-06-07 at 12 41 17 AM" src="https://github.com/arthurfiorette/axios-cache-interceptor/assets/52783389/aeadef6c-58e6-4cf0-9f6a-56f57924c05f">


Version 1.1.1
<img width="577" alt="Screenshot 2023-06-07 at 12 47 43 AM" src="https://github.com/arthurfiorette/axios-cache-interceptor/assets/52783389/1fb9e14b-04b3-4f0f-89d3-ee4d3fe27bdf">

**Comment by arthurfiorette** [comment]
First of all, @subhradeepnag, you could've reopened #545 if things weren't clear for you.

Let me walk you step-by-step:

> detected concurrent request, waiting for it to finish

You did not used await between the two requests, so both of the promises are being ran in parallel, but only the first request is sent, the other one is waiting for it to finish.

> deferred rejected, requesting again.

The first request was rejected. A.K.A Axios returned an axios error, probably because of a non 200 status code.  This way, one request was rejected, all left to us is to try again with the second one.

> cache predicate rejected this response

The second request was rejected by the cache predicate _(my bad for writing header interpreter on the previous github issue, i meant the same thing as in your logs: cache predicate)_. You can see the default behavior of `cachePredicate` in the [docs](https://axios-cache-interceptor.js.org/config/request-specifics#cache-cachepredicate).

Axios v0.10.7 worked with you because (as stated on axios-cache-interceptor releases), only at [ v1.1.0](https://github.com/arthurfiorette/axios-cache-interceptor/releases/tag/v1.1.0) the correct RFC behavior was added by default.

Again, my bad for writing header interpreter on the previous github issue, I meant the same thing as in your logs: cache predicate. But the answer was in your logs.

**Comment by arthurfiorette** [comment]
I guess should be fair to state in the [docs](https://axios-cache-interceptor.js.org/config/request-specifics#cache-cachepredicate) that these defaults are there to follow rfc 7231, as mentioned in the [source code](https://github.com/arthurfiorette/axios-cache-interceptor/blob/2d2696dadee4115f094efd2a6ef910f8fb1e4c38/src/cache/create.ts#L70). Up to a PR?

**Comment by subhradeepnag** [comment]
I've raised a PR - https://github.com/arthurfiorette/axios-cache-interceptor/pull/547
